File: src/main/java/xyz/oribuin/skyblock/managers/island/Island.java
package xyz.oribuin.skyblock.managers.island;

import org.bukkit.Location;

import java.util.UUID;

public final class Island {
    private final Location center;
    private boolean locked;
    private String name;
    private UUID owner;
    private int islandRange;
    private Location spawnPoint;

    public Island(String name, Location location, UUID owner, int islandRange) {
        this.name = name;
        this.owner = owner;
        this.islandRange = islandRange;
        this.center = location;
        this.spawnPoint = center;
        this.locked = false;
    }

    public int getIslandRange() {
        return this.islandRange;
    }

    public boolean isLocked() {
        return locked;
    }

    public String getName() {
        return this.name;
    }

    public UUID getOwner() {
        return this.owner;
    }

    public Location getCenter() {
        return this.center;
    }

    public Location getSpawnPoint() {
        return this.spawnPoint;
    }
}

File: Plugin/src/main/kotlin/xyz/oribuin/skyblock/island/Island.kt
package xyz.oribuin.skyblock.island

import org.bukkit.Location
import java.util.*

data class Island(var id: Int, var owner: UUID, var location: Location, var name: String) {
    var size = 200
    var spawnPoint = location
    var locked = false
    var members = mutableListOf<Member>()
    var coops = mutableListOf<UUID>()
}
